### PR TITLE
Add POST /tokens — Token creation

### PR DESCRIPTION
**Pull request by Natim**

### CODE REVIEW

File: conf/tests.ini [hunk 1]
```diff
@@ -3,11 +3,9 @@ use = egg:daybed
 daybed.backend = daybed.backends.memory.MemoryBackend
 backend.db_host = http://localhost:5984/
 backend.db_name = daybed-test
-persona.secret = THIS IS SUPER SECRET.
-persona.audiences = http://0.0.0.0:8000
 
 daybed.id_generator = daybed.backends.id_generators.UUID4Generator
-daybed.default_policy = read-only
+daybed.hawkHmacKey = c84035e13adf257e4b8ee9cdcae5022c
```
Comment (almet) [inline conf/tests.ini:8]: makes me think we probably want to update the docker file as well.

File: daybed/acl.py [hunk 2]
```diff
@@ -152,15 +172,15 @@ def __init__(self, request):
...
 
 
-def build_user_principals(user, request):
-    return [user]
+def build_user_principals(token, request):
+    return [token]
 
 
-def check_api_token(username, password, request):
+def check_api_token(hawkId, password, request):
```
Comment (leplatrem) [inline daybed/acl.py]: `token` ? 

I wouldn't spread the notion of hawk outside app init

File: daybed/backends/couchdb/__init__.py [hunk 3]
```diff
@@ -155,30 +155,27 @@ def delete_model(self, model_id):
         self._db.delete(doc)
         return doc
 
-    def __get_user(self, username):
+    def __get_token(self, hmacId):
         try:
```
Comment (leplatrem) [inline daybed/backends/couchdb/__init__.py:159]: token instead of hmacId ?

File: daybed/backends/memory/__init__.py [hunk 4]
```diff
@@ -122,31 +118,26 @@ def delete_model(self, model_id):
         del self._db['models'][model_id]
         return doc
 
-    def __get_user(self, username):
+    def __get_token(self, hmacId):
```
Comment (leplatrem) [inline daybed/backends/memory/__init__.py]: same same :)

File: daybed/acl.py (hunk 2 above)
Comment (Natim) [inline daybed/acl.py]: `tokenId` and `tokenKey` they?

File: daybed/backends/couchdb/__init__.py (hunk 3 above)
Comment (Natim) [inline daybed/backends/couchdb/__init__.py:159]: No it is not the `token`, it is the `tokenHmacId`

File: conf/tests.ini (hunk 1 above)
Comment (Natim) [inline conf/tests.ini:8]: Done

File: daybed/schemas/validators.py [hunk 5]
```diff
@@ -123,3 +123,31 @@ def model_validator(request):
         for record in records:
             validate_against_schema(request, definition_schema, record)
             request.validated['records'].append(record)
+
+
+def token_validator(request):
+    """Verify that the model is okay (that we have the right fields) and
```
Comment (almet) [inline daybed/schemas/validators.py]: that the model is valid.

File: daybed/schemas/validators.py [hunk 6]
```diff
@@ -123,3 +123,31 @@ def model_validator(request):
...
+    eventually populates it if there is a need to.
+    """
+    try:
+        body = json.loads(request.body.decode('utf-8'))
+    except ValueError:
+        request.errors.add('body', 'json value error', "body malformed")
```
Comment (almet) [inline daybed/schemas/validators.py]: "malformed body"

File: daybed/tests/test_views.py [hunk 7]
```diff
@@ -271,3 +268,18 @@ def test_record_deletion(self):
         # Test 404
         self.app.delete('/models/test/records/%s' % record_id,
                         headers=self.headers, status=404)
+
+
+class TokensViewsTest(BaseWebTest):
+
+    def test_post_token(self):
+        response = self.app.post('/tokens',
+                                 status=201)
```
Comment (almet) [inline daybed/tests/test_views.py]: put that on the same line

File: daybed/views/tokens.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+tokens = Service(name='tokens', path='/tokens', description='Tokens',
+                 renderer="jsonp", cors_origins=('*',))
+
+
+token = Service(name='token',
+                path='/tokens/{token_id}',
+                description='Token',
+                renderer="jsonp",
+                cors_origins=('*',))
```
Comment (almet) [inline daybed/views/tokens.py:13]: I'm not sure if that's js harming my judgment but I would expect to have these on a new line.

File: daybed/views/tokens.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+
+@tokens.post(permission='post_token')
+def post_tokens(request):
+    """Creates a new token and store it"""
+    session_token = os.urandom(32)
+    keyInfo = 'identity.mozilla.com/picl/v1/sessionToken'
```
Comment (almet) [inline daybed/views/tokens.py]: we probably should abstract this thing in a `get_hawk_session_token` function.

Comment (almet) [inline daybed/views/tokens.py]: We also need to comment that this url (identity.m.c/picl/blah) is the name of the scheme we use.

**Comment by almet** [comment]
Looks good to me. I will add hawk management once merged.

**Pull request merged**